test/service/__init__.py

```diff
@@ -21,16 +21,19 @@ def lambda_name(cls) -> str:
         return "service"
 
     @classmethod
-    def setUpClass(cls):
-        super().setUpClass()
-        for bundle in cls.bundles:
-            cls._index_canned_bundle(bundle)
+    def tearDownClass(cls):
```

Comment:
I understand that `clear_indices()` is idempotent but the asymmetry makes my head hurt. How many `tearDownClass` overrides do we save by doing this? And do those savings outweigh the potential confusion of seeing the effects of two consecutive invocations of `clear_indices()`. The pattern is "What you create in setUp, you destroy in tearDown. That symmetry should be maintained across all tests. Following that pattern slavishly adds sanity and predictability, if at the expense of mild increase in verbosity.

Reply:
I just counted 7 places where tearDownClass would need to be overridden but isn't because tearDownClass is is a part of WebServiceTestCase.

Reply:
To me that justifies keeping it there. What do you think?

Reply:
To me it doesn't. If suddenly that number dropped to five would we then remove the method? Three? We probably wouldn't because we'd simply forget. The surprising side-effect of performing the index deletion twice would remain. Also, say we left `tearDownClass()` here and had to override it in a subclass for some reason, would we rely on `super().tearDownClass()` calling `_teardown_indices()` or would we call it explicitly? In a `setupClass()` override we couldn't rely on `super.setupClass()` to call `_setup_indices()` for us. So we'd have to be asymmetric to be efficient (indices deleted once) or symmetric and inefficient (one redundant index deletion). I prefer to have symmetry and efficiency at the expense of some code duplication.

Reply:
OK. I'm convinced.

Reply:
Slightly embarrassed to admit it, but it probably doesn't need to be duplicated at all.. it looks like it can just be removed.